contentpal/src/test/java/org/dmfs/android/contentpal/testing/contentoperationbuilder/WithValues.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.android.contentpal.testing.contentoperationbuilder;
+
+import android.content.ContentProviderOperation;
+import android.content.ContentValues;
+
+import org.dmfs.optional.NullSafe;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeDiagnosingMatcher;
+
+import java.lang.reflect.Field;
+
+import static org.dmfs.android.contentpal.testing.contentvalues.Size.withValueCount;
+import static org.hamcrest.Matchers.allOf;
+
+
+/**
+ * A {@link Matcher} which matches a {@link ContentProviderOperation.Builder}.
+ *
+ * @author Marten Gajda
+ */
+public final class WithValues extends TypeSafeDiagnosingMatcher<ContentProviderOperation.Builder>
+{
+    private final Matcher<ContentValues> mValueMatcher;
+
+
+    @SafeVarargs
+    public static WithValues only(Matcher<ContentValues>... valueMatchers)
+    {
+        return new WithValues(allOf(allOf(valueMatchers), withValueCount(valueMatchers.length)));
+    }
+
+
+    public static WithValues withoutValues()
+    {
+        return new WithValues(withValueCount(0));
+    }
+
+
+    public WithValues(Matcher<ContentValues> valueMatchers)
+    {
+        mValueMatcher = valueMatchers;
+    }
+
+
+    @Override
+    protected boolean matchesSafely(ContentProviderOperation.Builder builder, Description mismatchDescription)
+    {
+        // create a builder an test the ContentValues
+        try
+        {
+            Field valuesField = ContentProviderOperation.Builder.class.getDeclaredField("mValues");
+            valuesField.setAccessible(true);
+
+            ContentValues values = new NullSafe<>((ContentValues) valuesField.get(builder)).value(new ContentValues());
+
+            if (!mValueMatcher.matches(values))
+            {
+                mValueMatcher.describeMismatch(values, mismatchDescription);
```

Comment:
It's repeated 3 times, because there are multiple `allOf` matchers involved. Definitely not ideal, but given the simplicity of the classes we're testing here, it's actually quite unlikely that you'll get many errors that you would have to decipher. So at present I'm not overly concerned.

At present I'd rather not reinvent Hamcrest. We're already doing a lot of other groundwork and we need to move forward with our actual projects.
The Hamcrest Matcher interface is not very complex and quite easy to implement. If you want to you can always throw an Exception. That's not how it's meant to be used but it would work.